alicloud/resource_alicloud_polardb_backup_policy.go

```diff
@@ -0,0 +1,136 @@
+package alicloud
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"strings"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/polardb"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudPolarDBBackupPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudPolarDBBackupPolicyCreate,
+		Read:   resourceAlicloudPolarDBBackupPolicyRead,
+		Update: resourceAlicloudPolarDBBackupPolicyUpdate,
+		Delete: resourceAlicloudPolarDBBackupPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cluster_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+
+			"backup_period": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{Type: schema.TypeString},
+				// terraform does not support ValidateFunc of TypeList attr
+				// ValidateFunc: validateAllowedStringValue([]string{"Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"}),
+				Optional: true,
+				Computed: true,
+			},
+
+			"backup_time": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice(BACKUP_TIME, false),
+				Optional:     true,
+				Default:      "02:00Z-03:00Z",
+			},
+		},
+	}
+}
+
+func resourceAlicloudPolarDBBackupPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+
+	d.SetId(d.Get("cluster_id").(string))
+
+	return resourceAlicloudPolarDBBackupPolicyUpdate(d, meta)
+}
+
+func resourceAlicloudPolarDBBackupPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	polardbService := PolarDBService{client}
+	object, err := polardbService.DescribeBackupPolicy(d.Id())
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+
+	d.Set("cluster_id", d.Id())
+	d.Set("retention_period", object.BackupRetentionPeriod)
+	d.Set("backup_period", strings.Split(object.PreferredBackupPeriod, ","))
+	d.Set("backup_time", object.PreferredBackupTime)
+
+	return nil
+}
+
+func resourceAlicloudPolarDBBackupPolicyUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*connectivity.AliyunClient)
+	polardbService := PolarDBService{client}
+	update := false
+
+	periodList := expandStringList(d.Get("backup_period").(*schema.Set).List())
+	backupPeriod := fmt.Sprintf("%s", strings.Join(periodList[:], COMMA_SEPARATED))
+	backupTime := d.Get("backup_time").(string)
+
+	if d.HasChange("backup_period") {
+		update = true
+	}
+
+	if d.HasChange("backup_time") {
+		update = true
+	}
+
+	if update {
```

Comment:
if d.HasChange("backup_period") || if d.HasChange("backup_time") {
		call the modify method
	}